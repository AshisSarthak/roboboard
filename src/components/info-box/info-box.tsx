import { Box, List, ListItem } from "@mui/material";
import { INSTRUCTIONS } from "../../utils/constants";

export const InfoBox = () => {
  return (
    <Box className="instructions">
      <header className="instruction-header">Instruction</header>
      <List>
        {INSTRUCTIONS.map((instruction: string, index: number) => {
          return <ListItem key={index}>{instruction}</ListItem>;
        })}
      </List>
    </Box>
  );
};
