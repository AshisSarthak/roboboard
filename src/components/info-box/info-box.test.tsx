import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { InfoBox } from "./info-box";

test("rendersInstruction", () => {
  const { asFragment } = render(<InfoBox />);
  expect(screen.getByText("Instruction")).toBeInTheDocument();
  expect(asFragment()).toMatchSnapshot();
});
