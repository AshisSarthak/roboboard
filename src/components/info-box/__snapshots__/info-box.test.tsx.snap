// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendersInstruction 1`] = `
<DocumentFragment>
  <div
    class="instructions MuiBox-root css-0"
  >
    <header
      class="instruction-header"
    >
      Instruction
    </header>
    <ul
      class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
    >
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        PLACE X,Y,F - will put the toy robot on the table in position X,Y and facing NORTH, SOUTH, EAST or WEST.
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        MOVE - will move the toy robot one unit forward in the direction it is currently facing.
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        LEFT | RIGHT - will rotate the robot 90 degrees in the specified direction without changing the position of the robot.
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
      >
        REPORT - will announce the X,Y and F of the robot. This can be in any form, but standard output is sufficient.
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
